en/docs/security-announcements/cve-justifications/2025/CVE-2025-46392.md

```diff
@@ -0,0 +1,30 @@
+---
+title: "CVE-2025-46392"
+category: security-announcements
+date: July 31 2025
+---
+
+# CVE-2025-46392
+
+<p class="doc-info">WSO2 Products impacted: no</p>
+<p class="doc-info">Customers actions required: no</p>
+---
+
+### REPORTED VULNERABILITY
+
+A Denial of Service (DoS) vulnerability in Apache Commons Configuration 1.x which can be exploited by supplying a specially crafted configuration file containing malicious or recursive variable interpolation, leading to uncontrolled resource consumption [^1].
+
+### REPORTED PRODUCTS
+
+- WSO2 Identity Server : 7.0.0, 7.1.0
+
+### WSO2 JUSTIFICATION
+
+Upon evaluating the impact of upgrading Apache Commons Configuration from version 1.x to 2.x, we identified significant architectural changes that introduce compatibility issues. Version 1.x uses the org.apache.commons.configuration.* package hierarchy, whereas version 2.x has been restructured under org.apache.commons.configuration2.*, with major modifications. Additionally, to make commons-configuration2 function as an OSGi service within the product, several new dependencies must be introduced, increasing the integration complexity. Due to these concerns, we are publishing this CVE justification with a detailed analysis of the CVE, outlining how associated risks are mitigated in WSO2 products.
+
+The reported vulnerability involves uncontrolled resource consumption via malicious or recursive variable interpolation in configuration files. In WSO2 Identity Server, the affected component is used exclusively to load the `log4j2.properties` file. This file is not user-controllable in deployment scenarios.
+
+Therefore, unless an attacker has write access to the server’s filesystem or configuration directories, the vulnerability is not exploitable in practice. Assuming proper server hardening and file system access controls, this issue poses no practical risk.
```

Comment:
@ayomawdb @JeethJJ Do we also need to say that we will address this in the upcoming release?

Reply:
My impression was that, since its a justification for the CVE, we do not necessarily need to mention that we will fix in later versions. We can add if necessary. Lets get the final call from @ayomawdb

Reply:
Up to the product to decide this, but there must be a roadmap item for these in the product side to make sure we do track these in future. We will be highlighting these issues again in future pre-release scans and at that time also pointing to this CVE justification is not right. Hence moving from Apache Commons Configuration 1.x to Apache Commons Configuration 2.x should be a roadmap item for next major release, internally.

Reply:
@JeethJJ @DMHP Please update once this done.

Reply:
Please find the roadmap item at [1]

[1] https://github.com/wso2/product-is/issues/24947